File: sdkwork-app-sdk-typescript/src/types/image-detail-vo.ts
import type { AuthorInfo } from './author-info';
import type { ImageMediaResource } from './image-media-resource';

/** 图片详情响应 */
export interface ImageDetailVO {
  /** 创建时间 */
  createdAt?: string;
  /** 更新时间 */
  updatedAt?: string;
  /** 图片ID */
  imageId?: string;
  /** 标题 */
  title?: string;
  /** 描述 */
  description?: string;
  /** URL */
  url?: string;
  /** 格式 */
  format?: string;
  /** 宽度 */
  width?: number;
  /** 高度 */
  height?: number;
  /** 比例 */
  aspectRatio?: string;
  /** 文件大小 */
  fileSize?: number;
  /** 资源信息 */
  resource?: ImageMediaResource;
  /** 作者信息 */
  author?: AuthorInfo;
  /** 状态 */
  status?: string;
  /** 是否公开 */
  isPublic?: boolean;
  /** 是否收藏 */
  isFavorite?: boolean;
  /** 查看次数 */
  viewCount?: number;
  /** 下载次数 */
  downloadCount?: number;
  /** 点赞数 */
  likeCount?: number;
}

File: sdkwork-app-sdk-go/types/image_detail_vo.go
package types

// 图片详情响应
type ImageDetailVO struct {
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
	ImageId string `json:"imageId"`
	Title string `json:"title"`
	Description string `json:"description"`
	Url string `json:"url"`
	Format string `json:"format"`
	Width int `json:"width"`
	Height int `json:"height"`
	AspectRatio string `json:"aspectRatio"`
	FileSize int `json:"fileSize"`
	Resource ImageMediaResource `json:"resource"`
	Author AuthorInfo `json:"author"`
	Status string `json:"status"`
	IsPublic bool `json:"isPublic"`
	IsFavorite bool `json:"isFavorite"`
	ViewCount int `json:"viewCount"`
	DownloadCount int `json:"downloadCount"`
	LikeCount int `json:"likeCount"`
}
